route-rs-runtime/src/link/mod.rs

```diff
@@ -33,7 +33,14 @@ mod blackhole_link;
 pub use self::blackhole_link::*;
 
 /// The type that all links take as input, or provide as output. This allows links to be combinatorial
-pub type PacketStream<Input> = Box<dyn Stream<Item = Input, Error = ()> + Send>;
+pub type PacketStream<Input> = Box<dyn Stream<Item = Input, Error = ()> + 'static + Send>;
+pub type TokioRunnable = Box<dyn Future<Item = (), Error = ()> + 'static + Send>;
+
+pub trait Link<A, B> {
+    fn new(ingress_streams: Vec<PacketStream<A>>) -> Self;
+    fn runnables(self) -> Vec<TokioRunnable>;
```

Comment:
Since `runnables` should be the last thing that's called on a `Link` I'm making it consume `self`. This is needed if we want to pass ownership of the object (which is also the `TokioRunnable`) to tokio.

Reply:
I think we are discussing this in the discord. While this is pretty good. Moving over to a builder syntax that build consumes self may be more ergonomic.  I assume that egressors would also need to consume self, since it will move the objects out as well.